tutor/hooks/filters.py

```diff
@@ -79,6 +79,13 @@ def add(
     ) -> t.Callable[
         [t.Callable[Concatenate[T, P], T]], t.Callable[Concatenate[T, P], T]
     ]:
+        """
+        Add a callback to the filter.
+
+        Callbacks are added by increasing priority. Highest priority score are called
+        last.
+        """
```

Comment:
Rather than write a new docstring for `Filter.add`, why not strike the docstring from module-level `add()` and adapt its contents here?

```suggestion
        """
        Decorator to add a callback function to be applied to this filter.
        :param int priority: optional order in which the filter callbacks are called. Higher
            values mean that they will be performed later. The default value is
            ``priorities.DEFAULT`` (10). Filters that should be called last should have a
            priority of 100.
        The return value of each filter function callback will be passed as the first argument to the next one.
        Usage::
            from tutor import hooks
            @hooks.Filters.MY_FILTER.add()
            def my_func(value, some_other_arg):
                # Do something with `value`
                ...
                return value
            # After filters have been created, the result of calling all filter callbacks is obtained by running:
            hooks.Filters.MY_FILTER.apply(initial_value, some_other_argument_value)
        """
```

Reply:
Yes, I agree. I moved all docstrings from the utility functions to their methods.